src/backend/aspen/api/utils/get_matching_gisaid_ids.py

```diff
@@ -0,0 +1,44 @@
+from typing import Iterable, Mapping, Set
+
+import sqlalchemy as sa
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from aspen.database.models import GisaidMetadata
+
+
+async def get_matching_gisaid_ids(
```

Comment:
This is the ~same code we have in `src/backend/aspen/app/utils.py` but updated for async db sessions and SqlAlchemy 2.0 query syntax